src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -168,73 +168,110 @@ private function lexValue()
-                if ('\\' === $this->data[$this->cursor] && isset($this->data[$this->cursor + 1]) && ('"' === $this->data[$this->cursor + 1] || "'" === $this->data[$this->cursor + 1])) {
+                ++$this->cursor;
+                $value = str_replace(array('\\\\', '\\"', '\r', '\n'), array('\\', '"', "\r", "\n"), $value);
+                $resolvedValue = $value;
+                $resolvedValue = $this->resolveVariables($resolvedValue);
+                $resolvedValue = $this->resolveCommands($resolvedValue);
+                $v .= $resolvedValue;
+            } else {
+                $value = '';
+                $prevChr = $this->data[$this->cursor - 1];
+                while ($this->cursor < $this->end && !in_array($this->data[$this->cursor], array("\n", '"', "'"), true) && !((' ' === $prevChr || "\t" === $prevChr) && '#' === $this->data[$this->cursor])) {
+                    if ('\\' === $this->data[$this->cursor] && isset($this->data[$this->cursor + 1]) && ('"' === $this->data[$this->cursor + 1] || "'" === $this->data[$this->cursor + 1])) {
+                        ++$this->cursor;
+                    }
+
+                    $value .= $prevChr = $this->data[$this->cursor];
+
+                    if ('$' === $this->data[$this->cursor] && isset($this->data[$this->cursor + 1]) && '(' === $this->data[$this->cursor + 1]) {
+                        ++$this->cursor;
+                        $value .= '('.$this->lexNestedExpression().')';
+                    }
+
                     ++$this->cursor;
                 }
+                $value = rtrim($value);
+                $resolvedValue = $value;
+                $resolvedValue = $this->resolveVariables($resolvedValue);
+                $resolvedValue = $this->resolveCommands($resolvedValue);
 
-                $value .= $prevChr = $this->data[$this->cursor];
-                ++$this->cursor;
+                if ($resolvedValue === $value && preg_match('/\s+/', $value)) {
+                    throw $this->createFormatException('A value containing spaces must be surrounded by quotes');
+                }
+
+                $v .= $resolvedValue;
+
+                if ($this->cursor < $this->end && '#' === $this->data[$this->cursor]) {
+                    break;
+                }
             }
-            $value = rtrim($value);
-        }
+        } while ($this->cursor < $this->end && "\n" !== $this->data[$this->cursor]);
 
         $this->skipEmptyLines();
 
-        $currentValue = $value;
-        if (!$singleQuoted) {
-            $value = $this->resolveVariables($value);
-            $value = $this->resolveCommands($value);
+        return $v;
+    }
+
+    private function lexNestedExpression()
+    {
+        ++$this->cursor;
+        $value = '';
+
+        while ("\n" !== $this->data[$this->cursor] && ')' !== $this->data[$this->cursor]) {
+            $value .= $this->data[$this->cursor];
+
+            if ('(' === $this->data[$this->cursor]) {
```

Comment:
https://github.com/xabbuh/symfony/blob/aaaf64ddde2aa22cb90a1c60fd902865589f5b18/src/Symfony/Component/Dotenv/Tests/DotenvTest.php#L144 and the line below trigger this part